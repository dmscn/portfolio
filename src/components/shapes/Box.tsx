import { ReactNode } from 'react'
import styled from 'styled-components'
// @ts-ignore
import { layout, color, flexbox, ColorProps } from 'styled-system'

const BaseBox = styled('div')(layout, flexbox)

const styles = {
  base: {
    display: 'flex',
  },
  // takes full viewport
  full: {
    minHeight: '100vh',
    maxWidth: '100vw',
  },
  // center the objects
  center: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  // flex column
  column: {
    flexDirection: 'column',
  },
}

export interface BoxProps {
  children?: ReactNode
  center?: boolean
  full?: boolean
  column?: boolean
}

type Props = BoxProps & ColorProps

const Box = styled(BaseBox).attrs((props: Props) => {
  const { center, full, column } = props
  return {
    ...styles.base,
    ...(full && styles.full),
    ...(center && styles.center),
    ...(column && styles.column),
  }
})<Props>`
  ${color}
`

export default Box
