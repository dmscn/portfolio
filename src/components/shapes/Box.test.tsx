import React from 'react'
import Box from './Box'
import { shallow } from 'enzyme'

describe('<Box />', () => {
  it('renders default', () => {
    const wrapper = shallow(<Box />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders full', () => {
    const wrapper = shallow(<Box full />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders center', () => {
    const wrapper = shallow(<Box center />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders default full center', () => {
    const wrapper = shallow(<Box full center />)
    expect(wrapper).toMatchSnapshot()
  })
})
