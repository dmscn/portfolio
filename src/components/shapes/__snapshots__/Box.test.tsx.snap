// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders center 1`] = `ShallowWrapper {}`;

exports[`<Box /> renders default 1`] = `ShallowWrapper {}`;

exports[`<Box /> renders default full center 1`] = `ShallowWrapper {}`;

exports[`<Box /> renders full 1`] = `ShallowWrapper {}`;
